doc/devel/contributing.rst

```diff
@@ -595,3 +595,52 @@ example code you can load it into a file handle with::
 
     import matplotlib.cbook as cbook
     fh = cbook.get_sample_data('mydata.dat')
+
+
+Baseline image generation
+-------------------------
+The first time this test is run, there will be no baseline image to compare against, so the test will fail. The baseline
+images would be generated either by installing ``matplotlib-baseline-images`` package or by running the baseline image
+generation command from the commandline. These methods are described below. Later on, developer can run ::
+
+    python3 -pytest
+
+and the tests will pass.
+
+
+How to install matplotlib-baseline-images package?
+--------------------------------------------------
+Developer can install the ``matplotlib-baseline-images`` package by running the command ::
+
+    python3 -mpip install  -ve  sub-wheels/matplotlib-baseline-images
+
+or can install the package from PyP by the command ::
+
+    python3 -mpip install matplotlib-baseline-images
+
+We will ensure cross-platform consistency (like we do now) by having these fixed set of "blessed" images sets (tied to
+a given release)
+
+
+Baseline image generation at the fresh install of matplotlib through command line
+----------------------------------------------------------------------------------
+The baseline images can be generated in the ``baseline_images`` directory by using ::
+
+    python3 -mpytest --generate_images
+
+For a given developer the baseline images don't change due to local changes (even if there may be some variation
+between developers respective baselines due to different freetype / latex etc.
+
+
+How to update an existing baseline image?
+---------------------------------------------
+Tests generating the baseline images may change over time. So, the baseline images needs to be regenerated by the
+developer after fetching the changes through ``git pull`` or ``git checkout -b featured_branch``. Missing baseline
+images can be generated in the ``baseline_images`` directory with the following command::
```

Comment:
So I still don't understand what will happen at this point.  I fix a bug, but it changes the image for  `test_boo.py::test_existing_feature`.   I decide the change is acceptable, and submit a PR.  How does the reviewer know that the image changed?  How do they compare the old versus new and assess if the new change is actually acceptable or not?  How do the CI tests pass?

Reply:
Great thoughts. I will write the documentation for those cases. Thanks